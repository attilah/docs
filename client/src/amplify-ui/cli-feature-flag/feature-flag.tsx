import {Component, Host, h, Prop, Element} from "@stencil/core";
import {hostStyle} from "./feature-flag.style";
import featureFlagsJson from "./feature-flags.json";

type FeatureFlags = Record<string, Section>;

type Section = {
  description: string;
  features: Record<string, FeatureFlag>;
};

export type FeatureFlag = {
  description: string;
  type: "Feature" | "Release" | "Experimental";
  valueType: "Boolean" | "Number" | "String";
  versionAdded: string;
  versionDeprecated?: string;
  deprecationDate?: string;
  versionRemoved?: string;
  removalDate?: string;
  values: Value[];
};

export type Value = {
  value: string;
  description: string;
  defaultNewProject: boolean;
  defaultExistingProject: boolean;
};

@Component({tag: "amplify-feature-flag", shadow: false})
export class AmplifyFeatureFlag {
  @Element() el: HTMLElement;

  render() {
    const data = featureFlagsJson as FeatureFlags;

    return (
      <Host
        class={{
          [hostStyle]: true,
        }}
      >
        {Object.entries(data).map(([name, section]) => {
          return (
            <div>
              <docs-in-page-link targetId={name}>
                <h2 id={name}>{name}</h2>
              </docs-in-page-link>

              {section.description ? <p>{section.description}</p> : undefined}

              {Object.entries(section.features).map(([flagName, flag]) => {
                return (
                  <amplify-feature-flag-summary
                    name={flagName}
                    feature={flag}
                  />
                );
              })}
            </div>
          );
        })}
      </Host>
    );
  }
}
