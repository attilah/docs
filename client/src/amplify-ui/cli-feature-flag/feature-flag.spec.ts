import {AmplifyFeatureFlag} from "./feature-flag";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-feature-flag", () => {
  describe("Component logic", () => {
    let featureFlag: AmplifyFeatureFlag;
    beforeEach(() => (featureFlag = new AmplifyFeatureFlag()));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyFeatureFlag],
            html: `<amplify-feature-flag />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
