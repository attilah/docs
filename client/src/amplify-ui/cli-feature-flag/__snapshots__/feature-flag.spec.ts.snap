// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-feature-flag Render logic should render 1`] = `
<amplify-feature-flag class="css-14qb6h6">
  <div>
    <docs-in-page-link targetid="graphQLTransformer">
      <h2 id="graphQLTransformer">
        graphQLTransformer
      </h2>
    </docs-in-page-link>
    <p>
      This section contains feature flags that are generic to the GraphQL transformation process itself and not tied to any particular transformer.
    </p>
    <amplify-feature-flag-summary name="transformerVersion"></amplify-feature-flag-summary>
  </div>
  <div>
    <docs-in-page-link targetid="keyTransformer">
      <h2 id="keyTransformer">
        keyTransformer
      </h2>
    </docs-in-page-link>
    <p>
      These feature flags are controlling \`@key\` transformer functionality.
    </p>
    <amplify-feature-flag-summary name="defaultQuery"></amplify-feature-flag-summary>
  </div>
</amplify-feature-flag>
`;
